client/API.md

```diff
@@ -0,0 +1,808 @@
+# Kapacitor API Reference Documentation
+
+Kapacitor provides an HTTP API on port 9092 by default.
+With the API you can control which tasks are executing, query status of tasks and manage recordings etc.
+
+Each section below defines the available API endpoints and there inputs and outputs.
+
+All requests are versioned and namespaced using the root path `/kapacitor/v1/`.
+
+### Response Codes
+
+All requests can return these response codes:
+
+| HTTP Response Code | Meaning                                                                                                                                                             |
+| ------------------ | -------                                                                                                                                                             |
+| 2xx                | The request was a success the content is dependent on the request.                                                                                                  |
+| 4xx                | Invalid request, refer to error for what it wrong with the request. Repeating the request will continue to return the same error.                                   |
+| 5xx                | The server was unable to process the request, refer to the error for a reason. Repeating the request may result in a success if the server issue has been resolved. |
+
+### Errors
+
+All requests can return JSON in the following format to provide more information about a failed request.
+
+```
+{
+    "error" : "error message"
+}
+```
+
+### Query Parameters vs JSON body
+
+To make using this API a consistent and easy experience we follow one simple rule for when extra information
+about a request is found in the query parameters of the URL or when they are part of the submitted JSON body.
+
+Query parameters are used only for GET requests and all other requests expect parameters to be specified in the JSON body.
+
+>NOTE: The /kapacitor/v1/write endpoint is the one exception to this rule since Kapacitor is compatible with the InfluxDB /write endpoint.
+
+## Writing Data
+
+Kapacitor can accept writes over HTTP using the line protocol.
+This endpoint is identical in nature to the InfluxDB write endpoint.
+
+| Query Parameter | Purpose                               |
+| --------------- | -------                               |
+| db              | Database name for the writes.         |
+| rp              | Retention policy name for the writes. |
+
+>NOTE: Kapacitor scopes all points by their database and retention policy. This means you MUST specify the `rp` for writes or Kapacitor will not know which retention policy to use.
+
+#### Example
+
```

Comment:
It will be nice to add reference to influxdb docs where they specify the format - http://docs.influxdata.com/influxdb/v0.12/write_protocols/line/

This endpoint should have precision, am I correct?